AWS/GCP: scylla-image-setup.service fails when no additional SSDs for RAID added on the instance

Seems like we have a bug to handle RAID for gcp, when no free disks available.
On the instance which added additional SSDs just works finely, so problem is no disk case.
Correct behavior for this is, detect no additional disks and skip RAID setup.

```
last failed login: Tue Mar 23 13:33:23 UTC 2021 from 35.235.240.66 on ssh:notty
There was 1 failed login attempt since the last successful login.

   _____            _ _       _____  ____  
  / ____|          | | |     |  __ \|  _ \ 
 | (___   ___ _   _| | | __ _| |  | | |_) |
  \___ \ / __| | | | | |/ _` | |  | |  _ < 
  ____) | (__| |_| | | | (_| | |__| | |_) |
 |_____/ \___|\__, |_|_|\__,_|_____/|____/ 
               __/ |                       
              |___/                        

Version:
       4.4.dev-0.20201208.f802356572
Nodetool:
        nodetool help
CQL Shell:
        cqlsh
More documentation available at: 
        http://www.scylladb.com/doc/
By default, Scylla sends certain information about this node to a data collection server. For information, see http://www.scylladb.com/privacy/

    Initial image configuration failed!

To see status, run 
 'systemctl status scylla-image-setup'

[syuu@takuya-scylla-test1 ~]$ systemctl status scylla-image-setup --no-pager -l
● scylla-image-setup.service - Scylla Cloud Image Setup service
   Loaded: loaded (/usr/lib/systemd/system/scylla-image-setup.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Tue 2021-03-23 13:33:25 UTC; 1min 17s ago
  Process: 943 ExecStart=/opt/scylladb/scylla-machine-image/scylla_image_setup (code=exited, status=1/FAILURE)
 Main PID: 943 (code=exited, status=1/FAILURE)

Mar 23 13:33:25 takuya-scylla-test1 scylla_image_setup[943]: AssertionError: No NVMe devices found!
Mar 23 13:33:25 takuya-scylla-test1 scylla_image_setup[943]: Traceback (most recent call last):
Mar 23 13:33:25 takuya-scylla-test1 scylla_image_setup[943]:   File "/opt/scylladb/scylla-machine-image/libexec/scylla_image_setup", line 27, in <module>
Mar 23 13:33:25 takuya-scylla-test1 scylla_image_setup[943]:     run('/opt/scylladb/scylla-machine-image/scylla_create_devices', shell=True, check=True)
Mar 23 13:33:25 takuya-scylla-test1 scylla_image_setup[943]:   File "/opt/scylladb/python3/lib64/python3.9/subprocess.py", line 524, in run
Mar 23 13:33:25 takuya-scylla-test1 scylla_image_setup[943]:     raise CalledProcessError(retcode, process.args,
Mar 23 13:33:25 takuya-scylla-test1 scylla_image_setup[943]: subprocess.CalledProcessError: Command '/opt/scylladb/scylla-machine-image/scylla_create_devices' returned non-zero exit status 1.
Mar 23 13:33:25 takuya-scylla-test1 systemd[1]: scylla-image-setup.service: Main process exited, code=exited, status=1/FAILURE
Mar 23 13:33:25 takuya-scylla-test1 systemd[1]: scylla-image-setup.service: Failed with result 'exit-code'.
Mar 23 13:33:25 takuya-scylla-test1 systemd[1]: Failed to start Scylla Cloud Image Setup service.

[syuu@takuya-scylla-test1 ~]$ sudo /opt/scylladb/scylla-machine-image/scylla_create_devices 
ls: cannot access '/dev/nvme0n*': No such file or directory
Traceback (most recent call last):
  File "/opt/scylladb/scylla-machine-image/libexec/scylla_create_devices", line 218, in <module>
    nvme_devices = get_nvme_devices()
  File "/opt/scylladb/scylla-machine-image/libexec/scylla_create_devices", line 195, in get_nvme_devices
    assert devices, "No NVMe devices found!"
AssertionError: No NVMe devices found!
```

--- a/common/scylla_create_devices
+++ b/common/scylla_create_devices
@@ -75,4 +75,7 @@ if __name__ == "__main__":
     else:
         disk_devices = get_disk_devices(instance, args.data_device)
 
-    create_raid(disk_devices)
+    if len(disk_devices) == 0:
+        print('No data disks found')
+    else:
+        create_raid(disk_devices)

--- a/common/scylla_image_setup
+++ b/common/scylla_image_setup
@@ -24,6 +24,10 @@ from subprocess import run
 if __name__ == '__main__':
     cloud_instance = get_cloud_instance()
     run('/opt/scylladb/scylla-machine-image/scylla_configure.py', shell=True, check=True)
+    if not os.path.ismount('/var/lib/scylla'):
+        print('Failed to initialize RAID volume, exiting setup')
+        sys.exit(1)
+
     run('/opt/scylladb/scripts/scylla_sysconfig_setup --nic eth0 --setup-nic --ami', shell=True, check=True)
     if cloud_instance.is_supported_instance_class():
         cloud_instance.io_setup()

